plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,511 @@
+    # Get BD
+    bds = [b.get("name") for b in schema_obj.get("templates")[template_idx]["bds"]]
+    if bd not in bds:
+        mso.fail_json(msg="Provided BD '{0}' does not exist. Existing BDs: {1}".format(bd, ", ".join(bds)))
+    return schema_obj.get("templates")[template_idx]["bds"][bds.index(bd)].get("uuid")
+
+
+def get_l3out_uuid(l3out_template_object, name):
+    l3outs = l3out_template_object.template.get("l3outTemplate", {}).get("l3outs", [])
+    match = l3out_template_object.get_object_by_key_value_pairs(
+        "L3Out",
+        l3outs,
+        [KVPair("name", name)],
+        fail_module=True,
+    )
+    if match:
+        return match.details.get("uuid")
+
+
+def get_ipsla_monitoring_policy_uuid(tenant_template_obj, uuid, name):
+    existing_ipsla_policies = tenant_template_obj.template.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaMonitoringPolicies", [])
+    match = tenant_template_obj.get_object_by_key_value_pairs(
+        "IPSLA Monitoring Policy",
+        existing_ipsla_policies,
+        [(KVPair("uuid", uuid) if uuid else KVPair("name", name))],
+        fail_module=True,
+    )
+    if match:
+        return match.details.get("uuid")
+
+
+def format_track_list_members(mso, mso_template, members, obj_cache):
+    track_list_members = []
+
+    def get_scope_obj_uuid(type, uuid, obj):
+        if uuid:
+            return uuid
+
+        name = obj.get("name")
+        template = obj.get("template")
+
+        if type == "bd":
+            schema_name = obj.get("schema")
+            if not schema_name:
+                mso.fail_json(msg="A member scope_type is bd and scope is used but the schema option is missing.")
+            key = "schema-{0}-{1}".format(type, schema_name)
+            schema_obj = obj_cache.get(key)
+            if not schema_obj:
+                id, path, schema_obj = mso.query_schema(schema_name)
+                obj_cache[key] = schema_obj
+            return get_bd_uuid(mso, schema_obj, template, name)
+
+        if type == "l3out":
+            key = "template-{0}-{1}".format(type, template)
+            mso_template = obj_cache.get(key)
+            if not mso_template:
+                mso_template = MSOTemplate(mso, type, template)
+                mso_template.validate_template(type)
+                obj_cache[key] = mso_template
+            return get_l3out_uuid(mso_template, name)
+
+    for member in members:
+        scope_type = member.get("scope_type")
+        track_member = {
+            "destIP": member.get("destination_ip"),
+            "scope": get_scope_obj_uuid(scope_type, member.get("scope_uuid"), member.get("scope")),
+            "scopeType": scope_type,
+            "ipslaMonitoringRef": get_ipsla_monitoring_policy_uuid(
+                mso_template,
+                member.get("ipsla_monitoring_policy_uuid"),
+                member.get("ipsla_monitoring_policy"),
+            ),
+        }
+        track_member = {"trackMember": track_member}
```

Comment:
why do this in a two step approach?
```suggestion
 track_member = {
    "trackMember": {
      "destIP": member.get("destination_ip"),
      "scope": get_scope_obj_uuid(scope_type, member.get("scope_uuid"), member.get("scope")),
      "scopeType": scope_type,
      "ipslaMonitoringRef": get_ipsla_monitoring_policy_uuid(
          mso_template,
          member.get("ipsla_monitoring_policy_uuid"),
          member.get("ipsla_monitoring_policy"),
      ),
    }
  }
```

Reply:
Changed